nrc_ws/src/nrc_nav/src/nrc_drive_pp.py

```diff
@@ -39,19 +39,26 @@ def generate_pure_pursuit_path():
         # add x,y coords from each point in the generated trajectory as waypoints.
         # this is better than just adding the 5 nodes as waypoints.
         pp.add_point(instructions[i][1], instructions[i][2])
-        # interpolate straight sections and fill with more points
-        if i < len(instructions) - 1:
-            x_gap = instructions[i][1] - instructions[i+1][1]
-            y_gap = instructions[i][2] - instructions[i+1][2]
-            density = 20
-            if abs(x_gap) > 2:
-                incr = x_gap / density
-                for n in range(density):
-                    pp.add_point(instructions[i][1] - n*incr, instructions[i][2])
-            elif abs(y_gap) > 2:
-                incr = y_gap / density
-                for n in range(density):
-                    pp.add_point(instructions[i][1], instructions[i][2] - n*incr)
+        #interpolate_path(i)
+
+def interpolate_path(i):
+    # interpolate straight sections and fill with more points
+    # this is useful for debugging and seeing the exact path
+    # However, does not work with the extra ramp waypoint
```

Comment:
What happens when this is used with the ramp waypoint?

Reply:
oh it 100% does not work with the ramp waypoint lol. I'll just do it the non-bodgy way and it should fix it.